package NKikimrTableStats;
option java_package = "ru.yandex.kikimr.proto";

message THistogramBucket {
    optional bytes Key = 1;
    optional uint64 Value = 2;
}

message THistogram {
    repeated THistogramBucket Buckets = 1;
}

message TChannelStats {
    optional uint32 Channel = 1;
    optional uint64 DataSize = 2;
    optional uint64 IndexSize = 3;
}

message TStoragePoolsStats {
    message TPoolUsage {
        optional string PoolKind = 1;
        optional uint64 DataSize = 2;
        optional uint64 IndexSize = 3;
    }
    repeated TPoolUsage PoolsUsage = 1;
}

message TTableStats {
    optional uint64 DataSize = 1; // both inMem and ondisk
    optional uint64 RowCount = 2; // both inMem and ondisk

    optional uint64 IndexSize = 3;
    optional uint64 InMemSize = 4;

    optional uint64 LastAccessTime = 5; // unix time in millisec
    optional uint64 LastUpdateTime = 6; // unix time in millisec

    optional THistogram RowCountHistogram = 7;
    optional THistogram DataSizeHistogram = 8;

    optional uint64 ImmediateTxCompleted = 9;
    optional uint64 PlannedTxCompleted = 10;
    optional uint64 TxRejectedByOverload = 11;
    optional uint64 TxRejectedBySpace = 12;
    optional uint64 TxCompleteLagMsec = 13;
    optional uint64 InFlightTxCount = 14;

    optional uint64 RowUpdates = 20;
    optional uint64 RowDeletes = 21;
    optional uint64 RowReads = 22;
    optional uint64 RangeReads = 23;
    optional uint64 RangeReadRows = 25;

    optional uint64 PartCount = 24;

    optional THistogram KeyAccessSample = 26;

    optional uint64 SearchHeight = 27;

    // seconds since epoch
    optional uint64 LastFullCompactionTs = 28;

    // i.e. this shard lent to other shards
    optional bool HasLoanedParts = 29;

    repeated TChannelStats Channels = 30;

    optional TStoragePoolsStats StoragePools = 31;

    optional uint64 ByKeyFilterSize = 32;
}
